### PR TITLE
Fix #5103: Add / Switch Network Dapps Request Integration

### LINKED ISSUE #5103: Add new EVM network
add-network

### PR DESCRIPTION
**Pull request by StephenHeaps**
## Summary of Changes
- Update `NetworkPicker` to use a Style struct, update `WalletPanelView` to re-use this `NetworkPicker` menu.
- Update `CustomNetworkDetails` to add a view-only mode (everything disabled)
- Integrated `PendingWebpageRequest.switchNetwork` and `PendingWebpageRequest.addNetwork` to open the `SuggestedNetworkView` modally using `WebpageRequestContainerView`.

This pull request fixes #5103

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
Note: At time of PR creation, wallet dapps are disabled with a compiler flag. Add -DWALLET_DAPPS_ENABLED to base.xcconfig to enable dapps support.

Network Picker in wallet panel: 
1. Visit a dapp site (ex. app.uniswap.org) and tap wallet icon in URL bar
2. Unlock wallet, click network in top right of panel

Switch Network:
1. Run [eth-manual-tests](https://github.com/bbondy/eth-manual-tests) locally
2. Click `wallet_switchEthereumChain` / `wallet_switchEthereumChain2`
Note: You may have to unlock you wallet before the modal to switch network appears
3. Tapping 'View Details' should show modal of network details, Tapping 'Cancel' should remove the pending request and dismiss, tapping 'Switch networks' should switch networks, cancel pending request and dismiss

Add Network:
1. Run [eth-manual-tests](https://github.com/bbondy/eth-manual-tests) locally
2. Click `wallet_addEthereumChain`
Note: You may have to unlock you wallet before the modal to switch network appears
3. Tapping 'View Details' should show modal of network details, Tapping 'Cancel' should remove the pending request and dismiss, tapping 'Approve' should add the network, cancel pending request and dismiss

## Screenshots:
network picker | switch | add
--|--|--

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by iccub — APPROVED** [review]
looks good to me, needs additional review by someone from ios-wallet team

File: BraveWallet/Settings/CustomNetworkDetailsView.swift
Comment (kylehickinson) [inline BraveWallet/Settings/CustomNetworkDetailsView.swift]: May want to concatenate `errorTitle` with description to make it more unique

File: BraveWallet/Settings/CustomNetworkDetailsView.swift
Comment (kylehickinson) [inline BraveWallet/Settings/CustomNetworkDetailsView.swift]: nit: `errMsg` → `message`/`errorMessage` (edit: scrolled down and saw it was just moved, but would still like the update 😄)

**Review by kylehickinson — APPROVED** [review]
Just some minor stuff if you want to pick up before merging :)

File: BraveWallet/Settings/CustomNetworkDetailsView.swift
Comment (StephenHeaps) [reply BraveWallet/Settings/CustomNetworkDetailsView.swift]: Fixed in [e1981c1](https://github.com/brave/brave-ios/pull/5164/commits/e1981c1f3b8a7fae4b404a4dd260caa3d45dd1c4)

File: BraveWallet/Settings/CustomNetworkDetailsView.swift
Comment (StephenHeaps) [reply BraveWallet/Settings/CustomNetworkDetailsView.swift]: Fixed in [e1981c1](https://github.com/brave/brave-ios/pull/5164/commits/e1981c1f3b8a7fae4b404a4dd260caa3d45dd1c4)

**Pull request merged**
